drivers/flash/flash_stm32_ospi.c

```diff
@@ -978,11 +978,149 @@ static int stm32_ospi_mem_reset(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_STM32_MEMMAP
+/* Function to configure the octoflash in MemoryMapped mode */
+static int stm32_ospi_set_memorymap(const struct device *dev)
+{
+	HAL_StatusTypeDef ret;
+	const struct flash_stm32_ospi_config *dev_cfg = dev->config;
+	struct flash_stm32_ospi_data *dev_data = dev->data;
+	OSPI_RegularCmdTypeDef s_command;
+	OSPI_MemoryMappedTypeDef s_MemMappedCfg;
+
+	/* Configure octoflash in MemoryMapped mode */
+	if ((dev_cfg->data_mode == OSPI_SPI_MODE) &&
+		(stm32_ospi_hal_address_size(dev) == HAL_OSPI_ADDRESS_24_BITS)) {
+		/* OPI mode and 3-bytes address size not supported by memory */
+		LOG_ERR("OSPI_SPI_MODE in 3Bytes addressing is not supported");
+		return -EIO;
+	}
+
+	/* Initialize the read command */
+	s_command.OperationType = HAL_OSPI_OPTYPE_READ_CFG;
```

Comment:
Will require some modification to make use of `read_opcode`, `read_dummy_cycles` in non DTR, OPI modes.

<details>

<summary>Example of a separate read cmd preparation function</summary>

```c


static int ospi_prepare_read_s_command(const struct device *dev, OSPI_Regulars_commandTypeDef *s_command)
{
	struct flash_stm32_ospi_data *dev_data = dev->data;
	const struct flash_stm32_ospi_config *dev_cfg = dev->config;

	s_command->OperationType = HAL_OSPI_OPTYPE_READ_CFG;
	s_command->FlashId = HAL_OSPI_FLASH_ID_1;
	/* default, may be overriden later */
	s_command->AlternateBytesMode = HAL_OSPI_ALTERNATE_BYTES_NONE;
	s_command->SIOOMode = HAL_OSPI_SIOO_INST_EVERY_CMD;

	if (dev_cfg->data_rate == OSPI_DTR_TRANSFER) {
		s_command->InstructionDtrMode = HAL_OSPI_INSTRUCTION_DTR_ENABLE;
		s_command->AddressDtrMode = HAL_OSPI_ADDRESS_DTR_ENABLE;
		s_command->DataDtrMode = HAL_OSPI_DATA_DTR_ENABLE;
		s_command->DQSMode = HAL_OSPI_DQS_ENABLE;
	}

	if (dev_cfg->data_mode == OSPI_OPI_MODE) {
		s_command->AddressSize = HAL_OSPI_ADDRESS_32_BITS;
		s_command->InstructionSize = HAL_OSPI_INSTRUCTION_16_BITS;

		s_command->InstructionMode = HAL_OSPI_INSTRUCTION_8_LINES;
		s_command->AddressMode = HAL_OSPI_ADDRESS_8_LINES;
		s_command->DataMode = HAL_OSPI_DATA_8_LINES;
	} else {
		s_command->AddressSize = stm32_ospi_hal_address_size(dev);
		s_command->InstructionSize = HAL_OSPI_INSTRUCTION_8_BITS;

		switch (dev_data->read_mode) {
		case JESD216_MODE_112: {
			s_command->InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;
			s_command->AddressMode = HAL_OSPI_ADDRESS_1_LINE;
			s_command->DataMode = HAL_OSPI_DATA_2_LINES;
			break;
		}
		case JESD216_MODE_122: {
			s_command->InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;
			s_command->AddressMode = HAL_OSPI_ADDRESS_2_LINES;
			s_command->DataMode = HAL_OSPI_DATA_2_LINES;
			break;
		}
		case JESD216_MODE_114: {
			s_command->InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;
			s_command->AddressMode = HAL_OSPI_ADDRESS_1_LINE;
			s_command->DataMode = HAL_OSPI_DATA_4_LINES;
			break;
		}
		case JESD216_MODE_144: {
			s_command->InstructionMode = HAL_OSPI_INSTRUCTION_1_LINE;
			s_command->AddressMode = HAL_OSPI_ADDRESS_4_LINES;
			s_command->DataMode = HAL_OSPI_DATA_4_LINES;
			break;
		}
		default:
			if (dev_cfg->data_mode == OSPI_QUAD_MODE) {
				cmd->InstructionMode = HAL_OSPI_INSTRUCTION_4_LINES;
				cmd->AddressMode = HAL_OSPI_ADDRESS_4_LINES;
				cmd->DataMode = HAL_OSPI_DATA_4_LINES;
				break;
			} else if (dev_cfg->data_mode == OSPI_DUAL_MODE) {
				cmd->InstructionMode = HAL_OSPI_INSTRUCTION_2_LINES;
				cmd->AddressMode = HAL_OSPI_ADDRESS_2_LINES;
				cmd->DataMode = HAL_OSPI_DATA_2_LINES;
				break;
			} else {
				return -ENOTSUP;
			}
		}
	}

	if (dev_cfg->data_rate == OSPI_DTR_TRANSFER) {
		/* DTR transfer rate (==> Octal mode) */
		s_command->Instruction = SPI_NOR_OCMD_DTR_RD;
		s_command->DummyCycles = SPI_NOR_DUMMY_RD_OCTAL_DTR;
	} else {
		/* STR transfer rate */
		if (dev_cfg->data_mode == OSPI_OPI_MODE) {
			/* OPI and STR */
			s_command->Instruction = SPI_NOR_OCMD_RD;
			s_command->DummyCycles = SPI_NOR_DUMMY_RD_OCTAL;
		} else {
			/* use SFDP:BFP read instruction */
			s_command->Instruction = dev_data->read_opcode;
			s_command->DummyCycles = dev_data->read_dummy_cycles;
		}
	}

	return 0;
}
```

</details>

Reply:
Ditto

Reply:
Opened #73840.